### PR TITLE
Adds cartocss to metadata of layergroup

### LINKED ISSUE #452: CartoCSS should be sent in metadata
We need this so that d3.cartodb is able to style maps

### PR DESCRIPTION
**Pull request by dgaubert**
Fixes #452

cc/ @rochoa @fdansv

### CODE REVIEW

File: lib/windshaft/metadata/layer-metadata.js [hunk 1]
```diff
@@ -47,6 +47,12 @@ LayerMetadata.prototype.getMetadata = function (rendererCache, params, mapConfig
             if (_.size(widgets)) {
                 metadata[layerId].widgets = widgets;
             }
+
+            var cartocss = mapConfig.getCartoCSS(layerId);
+
+            if (_.size(cartocss)) {
```
Comment (rochoa) [inline lib/windshaft/metadata/layer-metadata.js]: Why do we use `_` for this?

File: lib/windshaft/models/mapconfig.js [hunk 2]
```diff
@@ -167,6 +171,10 @@ MapConfig.prototype.getLayers = function() {
 };
 
 
+MapConfig.prototype.getCartoCSS = function(layerIndex) {
```
Comment (rochoa) [inline lib/windshaft/models/mapconfig.js]: Let's ship it like this, but I don't like the fact that we are exposing a function that only makes sense for some type of layers.

Comment (dgaubert) [inline lib/windshaft/models/mapconfig.js]: Makes sense, but I don't see any problem in that. I mean, it works like an abstract method, in some cases it applies and some other cases it doesn't. Probably I'm missing something.

File: lib/windshaft/metadata/layer-metadata.js (hunk 1 above)
Comment (dgaubert) [inline lib/windshaft/metadata/layer-metadata.js]: I've just copied `widgets` way. But yes, I don't like it.

**Pull request merged**